.github/workflows/docs.yml

```diff
@@ -35,6 +35,8 @@ jobs:
       VERIFY_MYPY_ERROR_CODES: 1
     steps:
       - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
```

Comment:
As far as I understand it, this is only necessary when using `actions/checkout` with `ssh` auth. Otherwise the `GITHUB_TOKEN` is used which is available anyway. I'd recommend not to add the `persist-credentials` setting.

https://github.com/actions/checkout/issues/485#issuecomment-934422611

Reply:
the rationale for the zizmor rule is given here: https://woodruffw.github.io/zizmor/audits/#artipacked (see also https://unit42.paloaltonetworks.com/github-repo-artifacts-leak-tokens/)

Are you suggesting we ignore the whole rule using a [configuration file](https://woodruffw.github.io/zizmor/configuration/)?

Reply:
Even if the token is persisted in an artifact, it's only valid for the workflow run and useless beyond that. It makes sense for ssh configs but for everything else it's just unnecessary IMO.

So yes, I'd ignore it here.

Reply:
Explicitly disabling the credential persisting doesn't do any _harm_ here though, right?

Maybe you're right and this is just cargo-culting, but zizmor's a tool written by a security expert, which I am not, so personally I'd be inclined to just go with what zizmor advises here. I don't have a strong opinion, though (again, I'm not a security expert 🙂)

Reply:
I would prefer to be secure by default. Don't see any actual profit in ignoring this rule.

Reply:
> Explicitly disabling the credential persisting doesn't do any _harm_ here though, right?

Not that I know of. It's probably just my personal preference for clean CI files 🙂